src/plasma/malloc.cc

```diff
@@ -129,41 +123,39 @@ int fake_munmap(void *addr, size_t size) {
   addr = pointer_retreat(addr, sizeof(size_t));
   size += sizeof(size_t);
 
-  struct mmap_record *record;
+  auto entry = mmap_records.find(addr);
 
-  HASH_FIND(hh_pointer, records_by_pointer, &addr, sizeof(addr), record);
-  if (record == NULL || record->size != size) {
+  if (entry == mmap_records.end() || entry->second.size != size) {
     /* Reject requests to munmap that don't directly match previous
      * calls to mmap, to prevent dlmalloc from trimming. */
     return -1;
   }
 
-  HASH_DELETE(hh_fd, records_by_fd, record);
-  HASH_DELETE(hh_pointer, records_by_pointer, record);
-
   int r = munmap(addr, size);
   if (r == 0) {
-    close(record->fd);
+    close(entry->second.fd);
   }
+
+  mmap_records.erase(entry);
```

Comment:
The erase replaces the `HASH_DELETE` that was a few lines up. It needed to be moved down, because the erase deallocates the entry, but we need to access `entry->second.fd`.